src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -1,56 +1,44 @@
 package com.atlassian.bitbucket.jenkins.internal.client;
 
 import com.atlassian.bitbucket.jenkins.internal.client.exception.*;
-import com.atlassian.bitbucket.jenkins.internal.config.BitbucketTokenCredentials;
 import com.atlassian.bitbucket.jenkins.internal.model.*;
-import com.cloudbees.plugins.credentials.Credentials;
-import com.cloudbees.plugins.credentials.common.UsernamePasswordCredentials;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import okhttp3.*;
-import org.apache.commons.codec.Charsets;
+import okhttp3.HttpUrl;
+import okhttp3.ResponseBody;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.log4j.Logger;
-import org.jenkinsci.plugins.plaincredentials.StringCredentials;
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.ConnectException;
-import java.net.SocketTimeoutException;
-import java.util.Base64;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
 
 import static com.atlassian.bitbucket.jenkins.internal.model.AtlassianServerCapabilities.WEBHOOK_CAPABILITY_KEY;
-import static java.net.HttpURLConnection.*;
 import static java.util.Collections.emptyMap;
 import static java.util.Collections.singletonMap;
 import static java.util.Objects.requireNonNull;
 import static okhttp3.HttpUrl.parse;
 
 public class BitbucketClientFactoryImpl implements BitbucketClientFactory {
 
-    private static final int BAD_REQUEST_FAMILY = 4;
-    private static final int SERVER_ERROR_FAMILY = 5;
     private static final Logger log = Logger.getLogger(BitbucketClientFactoryImpl.class);
+
     private final HttpUrl baseUrl;
-    private final Credentials credentials;
+    private final BitbucketCredentials credentials;
+    private HttpRequestExecutor httpRequestExecutor;
     private final ObjectMapper objectMapper;
-    private final Call.Factory httpCallFactory;
 
     BitbucketClientFactoryImpl(
             String serverUrl,
-            @Nullable Credentials credentials,
```

Comment:
Any reason for removing the `@Nullable` here?

Reply:
I have replaced Credentials with BitbucketCredentials which can never be null.

Reply:
Can we add a `requireNonNull` to the `BitbucketClientFactoryImpl` constructor